src/code-block-writer.ts

```diff
@@ -137,10 +137,15 @@ export default class CodeBlockWriter {
      * Conditionally writes a line of text.
      * @param condition - Condition to evaluate.
      * @param str - String to write if the condition is true.
-     */
-    conditionalWriteLine(condition: boolean | undefined, str: string) {
-        if (condition)
-            this.writeLine(str);
+     * Alternatively, a function which returns a string to write.
+     */
+    conditionalWriteLine(condition: boolean | undefined, str: string | (() => string)) {
+        if (condition) {
+            if (typeof str == 'string')
```

Comment:
We could reduce some code duplication by creating a reusable internal function (btw, best to do the test for function instead so that the else condition won't error when someone passes in something other than a function or string for some reason):

```ts
function getStringFromStrOrFunc(strOrFunc: string | (() => string)) {
    return strOrFunc instanceof Function ? strOrFunc() : strOrFunc;
}
```

Then using it:

```ts
this.write(getStringFromStrOrFunc(text));
```